pkg/controller/common.go

```diff
@@ -55,35 +66,65 @@ type CloudProvisioner interface {
 	GetMetricPrefix() string
 }
 
-const (
-	DriverName = "disk.csi.azure.com"
-)
+func cleanUpAzVolumeAttachmentByVolume(ctx context.Context, client client.Client, azClient azClientSet.Interface, namespace, azVolumeName string, mode cleanUpMode) error {
+	volRequirement, _ := labels.NewRequirement(VolumeNameLabel, selection.Equals, []string{azVolumeName})
+	labelSelector := labels.NewSelector().Add(*volRequirement)
 
-func CleanUpAzVolumeAttachment(ctx context.Context, client client.Client, azClient azClientSet.Interface, namespace, azVolumeName string) error {
-	var azVolume v1alpha1.AzVolume
-	err := client.Get(ctx, types.NamespacedName{Namespace: namespace, Name: azVolumeName}, &azVolume)
-	if err != nil {
-		klog.Errorf("failed to get AzVolume (%s): %v", azVolumeName, err)
+	attachments, err := azClient.DiskV1alpha1().AzVolumeAttachments(namespace).List(ctx, metav1.ListOptions{LabelSelector: labelSelector.String()})
+	if err != nil && !errors.IsNotFound(err) {
```

Comment:
Should we return from here if errors.IsNotFound(err) is true?

Reply:
Yes, that should allow us to skip unnecessary operations :)